### DISCUSSION TITLE
Astro Actions inspired handling of FormData

### DISCUSSION THREAD

**Discussion by maltesa** [next-safe-action/next-safe-action#160] [Ideas]
I really like how Astro Actions are abstracting Server Actions. They are also heavily inspired by trpc. For me, the handling of FormData looks like a good solution. I know that I could use `zod-form-data` for that, but I like the astro solution more.

- [Example](https://github.com/withastro/roadmap/blob/actions/proposals/0046-actions.md#handle-form-requests)
- [Form API Reference](https://github.com/withastro/roadmap/blob/actions/proposals/0046-actions.md#form-api-complete-reference )

Would a similar API be an idea for next-safe-action?

**Comment by TheEdoRan** [comment]
I personally think zod-form-data is better for this use case, but I'll keep this discussion open to see what the community thinks as well.

**Reply by maltesa** [reply]
Probably you're right. But in combination mit client-side validation (e.g. react-hook-form) I need two schemas, right? One for the client and one zod-form-data schema for the action. 
Ultimately, we want to validate the data of the same form. Using two schemas for it seems redundant. Or do you see a way to use the zod-form-data schema for react-hook-form etc.?

**Comment by alrightsure** [comment]
I 100% support this. No client-side form library works with zod-form-data, so I either need to duplicate schemas or only validate on the server.
